### PR TITLE
392 Capture error message from facets query

### PR DESCRIPTION
**Pull request by qifeng-bai**
#392

### CODE REVIEW

File: grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js [hunk 1]
```diff
@@ -130,16 +130,14 @@
...
-                                function(response){
-                                    if (response.status == 403 || response.status == 401) {
-                                        bootbox.alert($(i18n(539,"Authentication failed or login session expired, Please login again!")))
-                                    } else if (response.data.error) {
-                                        bootbox.alert("Error:" + response.data.error);
-                                    } else {
-                                        bootbox.alert("Unexpected error. Please check logs for more information")
-                                        console.log(response.data)
+                                function(error) {
+                                    if (!error.handled) {
```
Comment (nickdos) [inline grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js:134]: I can't see where `error.handled` is being set...
Also, where are 403 and 401 checks now being looked for? Its not clear from the code change....

File: grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js [hunk 2]
```diff
@@ -331,18 +325,20 @@
...
-                                $scope.errorMsg = error.data.error;
-                            }else {
-                                $scope.errorMsg = "Unexpected error. Check logs for more information";
+                        if (!error.handled) {
+                            if (error.status == 500) {
+                                $scope.errorMsg = "Unexpected error: the uploaded file may be broken or unrecognised.";
+                            } else {
+                                if (error.data.error) {
+                                    $scope.errorMsg = error.data.error;
+                                }else {
```
Comment (nickdos) [inline grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js:334]: needs a space before `else`

File: grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js (hunk 1 above)
Comment (qifeng-bai) [reply grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js:134]: There is an Http interceptor in spApp.js

https://github.com/AtlasOfLivingAustralia/spatial-hub/blob/392_facet_query/grails-app/assets/javascripts/spApp/spApp.js#L114

File: grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js (hunk 2 above)
Comment (qifeng-bai) [reply grails-app/assets/javascripts/spApp/controller/addAreaCtrl.js]: Changed

**Pull request merged**